pkg/status/combinedstatus-resolver.go

```diff
@@ -468,6 +507,41 @@ func (c *combinedStatusResolver) evaluateWorkStatusesPerBindingReadLocked(bindin
 	return combinedStatusesToQueue
 }
 
+// getObjectMetaAndSpec fetches the metadata and spec of the object associated
+// with the given workload object identifier.
+// The function is guaranteed not to return a key of `status` in the map.
+// If the resource contains any other subresources, they are fetched as well.
+func (c *combinedStatusResolver) getObjectMetaAndSpec(objectIdentifier util.ObjectIdentifier) (map[string]interface{}, error) {
+	// fetch object
+	lister, exists := c.wdsListers.Get(objectIdentifier.GVR())
+	if !exists {
+		return nil, fmt.Errorf("lister not found for gvr %s", objectIdentifier.GVR())
+	}
+
+	obj, err := lister.ByNamespace(objectIdentifier.ObjectName.Namespace).Get(objectIdentifier.ObjectName.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get object (%v) with gvr (%v): %w", objectIdentifier.ObjectName,
```

Comment:
If the workload object does not exist in the WDS then (a) it must have been recently deleted and (b) the user no longer wants CombinedStatus objects for it and (c) doing nothing but debug logging here will be adequate because deletion events are inbound, right?